Earthfile

```diff
@@ -159,6 +160,16 @@ go-build:
   SAVE ARTIFACT crossplane AS LOCAL _output/bin/${GOOS}_${GOARCH}/crossplane
   SAVE ARTIFACT crank AS LOCAL _output/bin/${GOOS}_${GOARCH}/crank
 
+# go-build-locally builds Crossplane binaries for the OS and architecture where
+# the earthly binary is invoked from.
+# This helps defining the correct OS/arch in cases where emulation is used, eg
+# by using the Docker desktop on Mac OS X. In this target though the OS/arch
+# can not be overriden by the --platform flag.
+go-build-locally:
+  LOCALLY
+  ARG TARGETPLATFORM
+  BUILD --platform=$TARGETPLATFORM +go-build
```

Comment:
i like this being the default for `earthly +build`, so that plain vanilla command will result with binaries for the host system that invoked the build 🤩

Reply:
There is a limitation though, when using `LOCALLY` we can not override the platform, so in a command similar to this `earthly --platform=windows/amd64 +go-build-locally` the --platform argument will be ignored.

Thus, the workflow for building locally the go binaries becomes the following:

1. For binaries with the same OS/arch as the host: `earthly +go-build-locally`
1. For binaries with the OS/arch as the emulation used in the host: `earthly +go-build`
1. For specifying the OS/arch: `earthly --platform=$OS/$ARCH +go-build`
1. For binaries on all available OS/archs: `earthly +go-multiplatform-build`

So if we add +go-build-locally in +build, it will become the same as 1. Are we fine with that?

Reply:
I don't think `LOCALLY` is doing anything here, is it?

`LOCALLY` means "run this build in my local environment, not in a container", but `BUILD` is going to run the `go-build` target in a container.

Reply:
ah thanks for clearing that up about the `LOCALLY` directive @negz - I agree that it would be best to continue running a containerized build, but do cross compilation when needed.

Reply:
correct, the `+go-build-locally` target was just setting the ARG and then passing the build to `go-build` which performs the actual build in a container. But either way, since the USERPLATFORM solution from above seems to work, I'm happily removing this target